### DISCUSSION TITLE
Is there any producer callback function available in librdkafka

### DISCUSSION THREAD

**Discussion by sayoojbk** [edenhill/librdkafka#3730] [Q&A]
Is there any event callback for producer-broker connection failure like we have for consumer-broker failure in APache Kafka like for Consumer we have KafkaRebalanceCallback? Does there exist something similar for the producer since my producer does not reconnect if my broker goes down and comes up again or shows some kinda log when this happens.

**Comment by edenhill** [comment]
Have a look at the error_cb  (`rd_kafka_conf_set_error_cb()`)
